src/snowflake/snowpark/session.py

```diff
@@ -1658,9 +1658,61 @@ def query_history(self) -> QueryHistory:
         return query_listener
 
     def _table_exists(self, table_name: str):
-        # TODO: Support qualified table names
-        tables = self._run_query(f"show tables like '{table_name}'")
-        return tables is not None and len(tables) > 0
+        # implementation based upon: https://docs.snowflake.com/en/sql-reference/name-resolution.html
+        validate_object_name(table_name)
+        # note: object name could have dots, e.g, a table could be created via: create table "abc.abc" (id int)
+        # currently validate_object_name does not allow it, but if in the future we want to support the case, we need to
+        # update the implementation accordingly in this method
+        try:
+            qualified_table_name = table_name.split(".")
+            if len(qualified_table_name) == 1:
+                # name in the form of "table"
+                tables = self._run_query(f"show tables like '{table_name}'")
+            elif len(qualified_table_name) == 2:
+                # name in the form of "schema.table" omitting database
+                # schema: qualified_table_name[0]
+                # table: qualified_table_name[1]
+                tables = self._run_query(
+                    f"show tables like '{qualified_table_name[1]}' in schema {qualified_table_name[0]}"
+                )
+            elif len(qualified_table_name) == 3:
+                # name in the form of "database.schema.table"
+                # database: qualified_table_name[0]
+                # schema: qualified_table_name[1]
+                # table: qualified_table_name[2]
+                if qualified_table_name[1] == "":
+                    # Double-Dot Notation: "database..table" in which the schema is empty
+                    tables = self._run_query(
+                        f"show tables like '{qualified_table_name[2]}' in database {qualified_table_name[0]}"
+                    )
+                else:
+                    # "database.schema.table", schema is not empty
+                    tables = self._run_query(
+                        f"show tables like '{qualified_table_name[2]}' in database {qualified_table_name[0]}"
```

Comment:
I think `show tables in schema database.schema` will also work, which is neater?

Reply:
it works! thanks for the suggestion!